docs/metrics-guide/averaging-methods.md

```diff
@@ -0,0 +1,60 @@
+# Averaging Methods
+
+## Description
+
+In the case of multi-class classification/object detection, you have to compute metrics such as precision, recall, and f1-score **per class**. When you want a single value that represents model performance across all classes, these per-class scores need to be aggregated. There are different averaging methods for doing this, namely, **macro**, **micro,** **weighted,** and **samples**.
+
+Let’s consider the following multi-class classification metrics:
+
+| Classes | TP | FP | FN | Precision | Recall | F1-score |
+| --- | --- | --- | --- | --- | --- | --- |
+| Airplane | 2 | 1 | 1 | 0.67 | 0.67 | 0.67 |
+| Boat | 1 | 3 | 0 | 0.25 | 1.0 | 0.4 |
+| Car | 3 | 0 | 3 | 1.0 | 0.5 | 0.67 |
+
+**Macro-averaging** is perhaps the most straightforward among the numerous options and is computed by taking an **unweighted** mean of all the per-class scores:
+
+```python
+macro-avg F1 = (0.67 + 0.4 + 0.67) / 3 = 0.58
+```
+
+In contrast, **micro-averaging** computes a **global** average by counting the sums of true positive (TP), false negative (FN) and false positive (FP):
+
+```python
+micro-avg F1 = 6 / (6 + 0.5 * (4 + 4)) = 0.6
+```
+
+But what about `micro-avg precision` and `micro-avg recall`?
```

Comment:
Would be nice to have sections (maybe `###` sub-headers) for each of these averaging types, such that they show up in the sidebar and it's clear what this page is about.

```
### Macro Average

...

### Micro Average

...

### Weighted Average

...
```